ACE/ace/SOCK_Dgram.cpp

```diff
@@ -746,39 +796,35 @@ ACE_SOCK_Dgram::make_multicast_ifaddr6 (ipv6_mreq *ret_mreq,
       // Initial call to determine actual memory size needed
       DWORD dwRetVal;
       ULONG bufLen = 0;
-      if ((dwRetVal = ::GetAdaptersAddresses (AF_INET6,
-                                              0,
-                                              0,
-                                              &tmp_addrs,
-                                              &bufLen)) != ERROR_BUFFER_OVERFLOW)
-        return -1; // With output bufferlength 0 this can't be right.
+      char *buf = 0;
+      if ((::GetAdaptersAddresses (AF_INET6,
+                                   0,
+                                   0,
+                                   &tmp_addrs,
+                                   &bufLen)) == ERROR_BUFFER_OVERFLOW)
+        {
+          ACE_NEW_RETURN (buf,
+                          char[bufLen],
+                          -1);
+        }
 
       // Get required output buffer and retrieve info for real.
-      PIP_ADAPTER_ADDRESSES pAddrs;
-      char *buf;
-      ACE_NEW_RETURN (buf,
-                      char[bufLen],
-                      -1);
-      pAddrs = reinterpret_cast<PIP_ADAPTER_ADDRESSES> (buf);
+      PIP_ADAPTER_ADDRESSES pAddrs = reinterpret_cast<PIP_ADAPTER_ADDRESSES> (buf);
       if ((dwRetVal = ::GetAdaptersAddresses (AF_INET6,
                                               0,
                                               0,
                                               pAddrs,
                                               &bufLen)) != NO_ERROR)
         {
-          delete[] buf; // clean up
```

Comment:
Leak of buf?

Reply:
It only needed to be deleted here because of the return on the following line. Since it will fall through to the final delete (line 790) now, there's no need to delete it here.